internal/provider/aws/provider_test.go

```diff
@@ -0,0 +1,4154 @@
+					Expect(input.Filters).To(ConsistOf(filters))
+					return &ec2.DescribeSecurityGroupsOutput{
+						SecurityGroups: []types.SecurityGroup{},
+					}, nil
+				})
+			err := p.ReconcileFirewallRulesDeletion(ctx, log, nodeName, "")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should return an error when an AWS API call (DescribeNetworkInterfaces) fails", func() {
+			mockec2Client.EXPECT().
+				DescribeSecurityGroups(ctx, gomock.AssignableToTypeOf(&ec2.DescribeSecurityGroupsInput{})).
+				DoAndReturn(func(_ context.Context, input *ec2.DescribeSecurityGroupsInput, _ ...func(*ec2.Options)) (*ec2.DescribeSecurityGroupsOutput, error) {
+					Expect(input.Filters).To(ConsistOf(filters))
+					return &ec2.DescribeSecurityGroupsOutput{
+						SecurityGroups: []types.SecurityGroup{
+							{
+								GroupId:   aws.String(groupID),
+								GroupName: aws.String(sgName),
+							},
+						},
+					}, nil
+				})
+			mockec2Client.EXPECT().
+				DescribeNetworkInterfaces(ctx, gomock.AssignableToTypeOf(&ec2.DescribeNetworkInterfacesInput{})).
+				DoAndReturn(func(_ context.Context, input *ec2.DescribeNetworkInterfacesInput, _ ...func(*ec2.Options)) (*ec2.DescribeNetworkInterfacesOutput, error) {
+					Expect(input.Filters).To(ConsistOf(MatchFields(IgnoreExtras, Fields{
+						"Name":   PointTo(Equal("group-id")),
+						"Values": ConsistOf(groupID),
+					})))
+					return &ec2.DescribeNetworkInterfacesOutput{
+						NetworkInterfaces: []types.NetworkInterface{},
+					}, fmt.Errorf("describe network interfaces error")
+				})
+
+			err := p.ReconcileFirewallRulesDeletion(ctx, log, nodeName, "")
+			Expect(err).To(MatchError(&provider.Error{
+				Code: "InternalError",
+				Msg:  "failed to list network interfaces: describe network interfaces error",
+			}))
+		})
+
+		It("should return an error when an AWS API call (ModifyNetworkInterfaceAttribute) fails", func() {
+			mockec2Client.EXPECT().
+				DescribeSecurityGroups(ctx, gomock.AssignableToTypeOf(&ec2.DescribeSecurityGroupsInput{})).
+				DoAndReturn(func(_ context.Context, input *ec2.DescribeSecurityGroupsInput, _ ...func(*ec2.Options)) (*ec2.DescribeSecurityGroupsOutput, error) {
+					Expect(input.Filters).To(ConsistOf(filters))
+					return &ec2.DescribeSecurityGroupsOutput{
+						SecurityGroups: []types.SecurityGroup{
+							{
+								GroupId:   aws.String(groupID),
+								GroupName: aws.String(sgName),
+							},
+						},
+					}, nil
+				})
+			mockec2Client.EXPECT().
+				DescribeNetworkInterfaces(ctx, gomock.AssignableToTypeOf(&ec2.DescribeNetworkInterfacesInput{})).
+				DoAndReturn(func(_ context.Context, input *ec2.DescribeNetworkInterfacesInput, _ ...func(*ec2.Options)) (*ec2.DescribeNetworkInterfacesOutput, error) {
+					Expect(input.Filters).To(HaveLen(1))
+					Expect(aws.ToString(input.Filters[0].Name)).To(Equal("group-id"))
+					Expect(input.Filters[0].Values).To(ConsistOf(groupID))
```

Comment:
You forgot to fix https://github.com/quortex/kubestatic/pull/186#discussion_r2060264221 here.